### PR TITLE
Support custom CA certificate bundles

### PR DESCRIPTION
**Pull request by kurtmckee**
Added
-----

* Support custom CA certificate bundles.

  Previously, SSL/TLS verification allowed only a boolean ``True`` or ``False`` value.
  It is now possible to specify a CA certificate bundle file
  using the existing ``verify_ssl`` parameter
  or ``GLOBUS_SDK_VERIFY_SSL`` environment variable.

  This may be useful for interacting with Globus through certain proxy firewalls.

----
:books: Documentation preview :books:: https://globus-sdk-python--903.org.readthedocs.build/en/903/

### CODE REVIEW

File: src/globus_sdk/config/env_vars.py [hunk 1]
```diff
@@ -70,12 +61,28 @@ def _load_var(
...
+        if value.lower() in {"n", "no", "f", "false", "off", "0"}:
+            return False
+        if os.path.isfile(value):
+            return value
+    if isinstance(value, pathlib.Path):
+        if value.is_file():
+            return str(value.absolute())
+    raise ValueError(
+        f"SSL verification {value} is not a valid boolean value "
+        "nor a path to a file that exists"
```
Comment (sirosen) [inline src/globus_sdk/config/env_vars.py:84]: Nit: not -> or, or neither -> nor; but not -> nor looks awkward to me. (I'm not certain about this being a correctness thing, but it reads in an unusual way.)

```suggestion
        "or a path to a file that exists"
```

File: src/globus_sdk/transport/requests.py [hunk 2]
```diff
@@ -162,8 +163,9 @@ def tune(
         In particular, this can be used to temporarily adjust request-sending minutiae
         like the ``http_timeout`` used.
 
-        :param verify_ssl: Explicitly enable or disable SSL verification
-        :type verify_ssl: bool, optional
+        :param verify_ssl: Explicitly enable or disable SSL verification,
+            or configure the path to a CA certificate bundle to use for SSL verification
+        :type verify_ssl: bool, str, or pathlib.Path, optional
```
Comment (sirosen) [inline src/globus_sdk/transport/requests.py:168]: This same doc update needs to be reflected in the `__init__` parameter docs in the class docstring.

File: src/globus_sdk/config/env_vars.py [hunk 3]
```diff
@@ -70,12 +61,28 @@ def _load_var(
...
+    if isinstance(value, str):
+        if value.lower() in {"y", "yes", "t", "true", "on", "1"}:
+            return True
+        if value.lower() in {"n", "no", "f", "false", "off", "0"}:
+            return False
+        if os.path.isfile(value):
+            return value
+    if isinstance(value, pathlib.Path):
+        if value.is_file():
+            return str(value.absolute())
```
Comment (sirosen) [inline src/globus_sdk/config/env_vars.py:81]: :+1: or :-1: on this adjustment?
```suggestion
    if isinstance(value, pathlib.Path) and value.is_file():
        return str(value.absolute())
```

**Review by sirosen — CHANGES_REQUESTED** [review]
Some very minor changes needed, but this looks good!

This has been an outstanding issue for a little while but I, for some reason, thought it would be more complex to implement support.

File: src/globus_sdk/config/env_vars.py (hunk 1 above)
Comment (kurtmckee) [reply src/globus_sdk/config/env_vars.py:84]: "Not" and "nor" are correct, but I can rephrase to something like

> SSL verification value must be a boolean value or a path to a file that exists

File: src/globus_sdk/transport/requests.py (hunk 2 above)
Comment (kurtmckee) [reply src/globus_sdk/transport/requests.py:168]: Oh! It's in the class docstring! It seemed weird to me that `.__init__` was completely undocumented. 😆

File: src/globus_sdk/config/env_vars.py (hunk 1 above)
Comment (kurtmckee) [reply src/globus_sdk/config/env_vars.py:84]: New rev coming

File: src/globus_sdk/transport/requests.py (hunk 2 above)
Comment (kurtmckee) [reply src/globus_sdk/transport/requests.py:168]: New rev coming

File: src/globus_sdk/config/env_vars.py (hunk 3 above)
Comment (kurtmckee) [reply src/globus_sdk/config/env_vars.py]: New rev.

**Review by sirosen — APPROVED** [review]

**Pull request merged**